### PR TITLE
CHI-1421: Unit tests & minor edge case fixes for resourceReferral state

### PR DESCRIPTION
**Pull request by stephenhand**
## Description

* Unit tests & minor edge case fixes for resourceReferral state

### CODE REVIEW

File: plugin-hrm-form/src/___tests__/states/contacts/resourceReferral.test.ts [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        ...baseState,
+        tasks: {
+          ...baseState.tasks,
+          task1: {
+            ...baseState.tasks.task1,
+            draft: {
+              ...baseState.tasks.task1.draft,
+              resourceReferralList: {
+                ...baseState.tasks.task1.draft.resourceReferralList,
+                resourceReferralIdToAdd: '1234',
```
Comment (mythilytm) [inline plugin-hrm-form/src/___tests__/states/contacts/resourceReferral.test.ts:66]: It is nice see this use of spread operator to copy the initial state and adding properties by test case

**Review by mythilytm — APPROVED** [review]
Did not test locally but this looks good.

File: plugin-hrm-form/src/___tests__/states/contacts/resourceReferral.test.ts (hunk 1 above)
Comment (stephenhand) [reply plugin-hrm-form/src/___tests__/states/contacts/resourceReferral.test.ts:66]: Yea it gets a little ugly when it's this deep into the structure
